scripts/ci/bundle_and_upload_rerun_cpp.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
+
+from __future__ import annotations
+
+import argparse
+import logging
+import multiprocessing
+import os
+import shutil
+import subprocess
+import tempfile
+
+from google.cloud import storage
+
+
+def run(
+    args: list[str], *, env: dict[str, str] | None = None, timeout: int | None = None, cwd: str | None = None
+) -> None:
+    print(f"> {subprocess.list2cmdline(args)}")
+    result = subprocess.run(args, env=env, cwd=cwd, timeout=timeout, check=False, capture_output=True, text=True)
+    assert (
+        result.returncode == 0
+    ), f"{subprocess.list2cmdline(args)} failed with exit-code {result.returncode}. Output:\n{result.stdout}\n{result.stderr}"
+
+
+def download_rerun_c(target_dir: str, git_hash: str, platform_filter: str = None) -> None:
+    logging.info("Downloading rerun_c...")
+
+    os.mkdir(target_dir)
+
+    # See reusable_build_and_upload_rerun_c.yml for available libraries.
+    # See rerun_cpp/CMakeLists.txt for library names.
+    for src, dst in [
+        ("linux/librerun_c.a", "librerun_c__linux_x64.a"),
+        ("macos-arm/librerun_c.a", "librerun_c__macos_arm64.a"),
+        ("macos-intel/librerun_c.a", "librerun_c__macos_x64.a"),
+        ("windows/librerun_c.a", "librerun_c__win_x64.a"),
+    ]:
+        if platform_filter is not None and src.startswith(platform_filter) is False:
+            continue
+
+        url = f"https://build.rerun.io/commit/{git_hash}/rerun_c/{src}"
+        run(["curl", "-L", "-o", f"{target_dir}/{dst}", url])
+
+
+def upload_rerun_cpp(rerun_zip: str, git_hash: str) -> None:
+    logging.info("Uploading to gcloud...")
+
+    gcs = storage.Client("rerun-open")
+    bucket = gcs.bucket("rerun-builds")
+    destination = bucket.blob(f"commit/{git_hash}/rerun_cpp.zip")
+    destination.content_type = "application/zip"
+    destination.upload_from_filename(rerun_zip)
+    logging.info(f"Uploaded to https://build.rerun.io/commit/{git_hash}/rerun_cpp.zip")
```

Comment:
nitpick, but I think we should call it `rerun_cpp_sdk.zip` for extra explicitness